src/raster/r.survey/r.survey.py

```diff
@@ -0,0 +1,683 @@
+    memory = int(options['memory'])
+    binary = flags['b']
+    hcurv = flags['c']
+    downward = flags['d']
+    # converting points map to 3d if a layer and elevation field is provided
+    try:
+        if obsabselev:
+            if layer:
+                Module("v.to.3d", input=pnt, output="xxtemppnt3d", type="point", column=obsabselev, layer=layer, quiet=True)
+                pnt = "xxtemppnt3d"
+    except:
+        message = "There was an error converting the layer to 3d. Plese check if you have provided column and layer information. Exiting "
+        gscript.error(message)
+        # cleaning termporary layers
+        cleanup()
+        sys.exit()
+    try:
+        oradius
+    except:
+        oradius = 0
+    # exporting some variables for other functions
+    main.treesmap = treesmap
+    main.buildmap = buildmap
+    main.nprocs = nprocs
+    try:
+        # setting the starting region alignement to the grid of the DEM
+        Module("g.region", align=dem)
+        # the following 2 lines are only to verify that there are no maps with the same names as output maps. To prevent overwrinting error after all the calculationsa have been done.
+        Module("r.mapcalc", expression="{A} = {B}".format(A=output+'_maxViewAngle', B=1), quiet=True)
+        Module("g.remove", type='raster', name=output+'_maxViewAngle', quiet=True, flags="f")
+        # running the "general" function
+        general(pnt, dem, treesmap, buildmap, treesheigh, treesheighdeviation, buildheigh, obsabselev)
+        # creating the pool ofr the parallel processes. initialization of the single processes is delayed by the lock
+        pool = multiprocessing.Pool(processes=nprocs, initializer=init, initargs=[multiprocessing.Lock()])
+        # Running the "iterate" function
+        iterate(pnt, general.dem, obs_heigh, maxdist, hcurv, downward, oradius, general.ctg, nprocs, obsabselev, memory)
+        # running the "compute" function in parallel
+        pool.starmap(compute, [(t) for t in iterate.tasks])
+        # terminating the parallel computation
+        pool.close()
+        # using the generated temporary maps to create the final maps
+        try:
+            # the following to split the categories in nprocs chunks
+            chunks = [general.ctg[x:x+nprocs] for x in range(0, len(general.ctg), nprocs)]
+            # creating the tasks for the parallel processing
+            tasks2 = list(zip(chunks, range(len(chunks))))
+            # creating the "zeros" map to be used for the combination of the different teporary maps  (THE FOLLOWING CAN BE DONE CREATING A FIRTS MAP AND THEN COPYING IT, FASTENING THE PROCESS)
+            for i in range(len(chunks)):
+                message = "Creating \"maps zero\" for the chunk %s  of %s"
+                gscript.message(message % (str(i), str(len(chunks))))
+                # for storing maximum view angles
+                Module("r.mapcalc", expression="xxtemp_a_{p} = 0".format(p=str(i)), quiet=True)
```

Comment:
There are functions in grass.script (grass.script.utils) to create unique names for temporary maps.

Reply:
Hi Vaclav. Thank you for the suggestion. Very interesting,
However in this case it is more simple, in our case, to keep things as they are. Is this a major problem or we have to change all the names in the code?
Thank you very much

Reply:
Yes. we will do that but, as I wrote in another comment, we have some initial lines of code that check if layers having the same names exists in the mapset. In that case the code stops and ask for rename or delete them...

Reply:
Right, it would be just for user convenience.